src/clients/UBAClient/UBAClientAbstract.ts

```diff
@@ -0,0 +1,139 @@
+import winston from "winston";
+import { RefundRequestWithBlock, UbaFlow } from "../../interfaces";
+import { BigNumber } from "ethers";
+import { UBAFeeSpokeCalculator } from "../../UBAFeeCalculator";
+
+export type RequestValidReturnType = { valid: false; reason: string } | { valid: true };
+export type OpeningBalanceReturnType = { blockNumber: number; spokePoolBalance: BigNumber };
```

Comment:
Minor: FWIW after naming it `spokePoolBalance` I was wondering if `tokenBalance` would be better.

Reply:
isn't it always going to refer to the balance of the spoke pool? If so, I would find `tokenBalance` confusing sometimes since it sounds like the wallet balance of an EOA, such as the relayer